src/include/sof/audio/buffer.h

```diff
@@ -47,7 +47,7 @@ extern struct tr_ctx buffer_tr;
 #define trace_buf_get_id(buf_ptr) ((buf_ptr)->pipeline_id)
 
 /** \brief Retrieves subid (comp id) from the buffer */
-#define trace_buf_get_subid(buf_ptr) ((buf_ptr)->id)
+#define buf_get_id(buf_ptr) ((buf_ptr)->stream.runtime_stream_params.id)
```

Comment:
is the ID really a runtime parameter? Isn't it persistent?

Reply:
it is set once, not modified later. Anyway - in rutime
As it is a common parameter for all streams, no matter how buffers are implemented, so I put it to this container
But you're right, it the name of the container may be somehow misleading, I'll change it